test/index.js

```diff
@@ -397,6 +397,48 @@ describe('Joi', () => {
         Helper.validate(Joi.compile({ other: Joi.number() }), [[{ foo: 'bar' }, false, '"foo" is not allowed']]);
     });
 
+    it('local unknown setting always overrides global allowUnknown setting', () => {
+        const input = {
+            a: {
+                b: "h",
+                c: "f"
+            },
+            d: "h"
+        };
+
+        Helper.validate(
+            Joi.object({
+                a: Joi.object().keys({
+                    b: Joi.string()
+                }).unknown(true)
+            }),
+            { allowUnknown: false },
+            [[input, false, '"d" is not allowed']]
+        );
+
+        Helper.validate(
+            Joi.object({
+                a: Joi.object().keys({
+                    b: Joi.string()
+                }).unknown(false)
+            }),
+            { allowUnknown: true },
+            [[input, false, '"a.c" is not allowed']]
+        );
+
+        Helper.validate(
+            Joi.object({
+                a: Joi.object().keys({
+                    b: Joi.string()
+                })
+            }),
+            { allowUnknown: true },
+            [[input, false, '"a.c" is not allowed']]
+        );
```

Comment:
@Marsup see this test, `allowUnknown` only applies to keys without ancestors

Reply:
Sorry for the late answer but I'm not sure I see your point. Joi should consider local options as higher priority than global options, it's not related to whether there are ancestors or not.

Reply:
>  Joi should consider local options as higher priority than global options

correct and this is what this diff aims to achieve

> it's not related to whether there are ancestors or not.

checking the ancestor is the only establish locality

```
if (has_ancestor) {
// ignore global setting and evaluate local unknown setting
} else {
// evaluate global setting
}
```